src/components/CompleteProfile/CompleteProfile.tsx

```diff
@@ -92,327 +175,512 @@ const CompleteProfile = () => {
 
 	const handleRemoveSocialAccount = (socialAccountToRemove: string) => {
 		setSocialAccountInputs((previousAccounts) => previousAccounts.filter((socialAccount: string) => socialAccount !== socialAccountToRemove));
+		setProfileData((prev) => {
+			const updatedLinks = prev.links.filter((link) => link.platform !== socialAccountToRemove);
+
+			return {
+				...prev,
+				links: updatedLinks
+			};
+		});
 		toggleIconVisibility(socialAccountToRemove);
 	};
 
+	const getUpdateProfileParams = (_formData: FormData): UpdateProfileParams => (
+		profileData as unknown as UpdateProfileParams
+	);
+
+	const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+		const { name, value } = e.target;
+		setProfileData((prev) => ({
+			...prev,
+			[name]: value
+		}));
+	};
+
+	const handleLinksInputChange = (platform: string, value: string) => {
+		const trimmedValue = value.trim();
+
+		setProfileData((prev) => {
+			const existingLink = prev.links.find((link) => link.platform === platform);
+
+			let updatedLinks;
+			if (existingLink) {
+				if (trimmedValue === '') {
+					updatedLinks = prev.links.filter((link) => link.platform !== platform);
+				} else {
+					updatedLinks = prev.links.map((link) => link.platform === platform ? { ...link, url: trimmedValue } : link);
+				}
+			} else if (trimmedValue !== '') {
+				updatedLinks = [...prev.links, { platform, url: trimmedValue }];
+			} else {
+				return prev;
+			}
+
+			return {
+				...prev,
+				links: updatedLinks
+			};
+		});
```

Comment:
```suggestion
		if (trimmedValue) {
			setProfileData((prev) => {
				const existingLinkIndex = prev.links.findIndex((link) => link.platform === platform);
				const updatedLinks = prev.links;
				
				if (existingLinkIndex !== -1) {
					updatedLinks.splice(existingLinkIndex, 1);
				}

				return {
					...prev,
					links: [...updatedLinks, {
						platform,
						value: trimmedValue
					}]
				};
			});
		}
```

1. Only update the value if there is anything to update
2. Using [`Array.findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) and [`Array.splice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) will remove the item from the array while preserving the order of items if needed. (Another option is to just filter the item out).
3. The spread operator + the item is an easy shortcut to appending items to an array. Don't abuse it though as it will iterate over all items and then append the new one.

Reply:
@madcampos I am updating the value when its empty because I want to remove the value if a user empties out the input field after initially typing. Otherwise the input will look like { platform: "LinkedIn", url: "" }.
This removes the respective object from the links array altogether.

Reply:
That is okay to have a sparse array of items. But then again: if we have a defined set of items, why not save them as an object with the format:
```typescript
Record<PlatformName = string, Url = string>
```

Reply:
We are not manipulating this state anymore with onChange.